### PR TITLE
First attempt at running on darwin aarch64

### PR DESCRIPTION
**Pull request by pugnascotia**
This PR is basically a bunch of hacks to get the build to run on an Apple M1 (ARM 64 / aarch64) machine. I think the changes are mostly reasonable, apart from some hard-coding to use the Azul JVM, which at the time of writing seems to be the only available JVM.

A basic `./gradlew run` fails because it can't run the ML subprocess, but if you disable that then ES starts just fine.

I'm not necessarily suggesting we merge this as-is, but I think @mark-vieira and @breskeby would be interested in what I did.

### CODE REVIEW

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/2

File: buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java [hunk 1]
```diff
@@ -131,6 +132,22 @@ private void setupRepository(Project project, Jdk jdk) {
                     + jdk.getBuild()
                     + "/GPL/openjdk-[revision]_[module]-[classifier]_bin.[ext]";
             }
+        } else if (jdk.getVendor().equals(VENDOR_AZUL)) {
+            repoUrl = "https://cdn.azul.com";
+
+            // The following is an absolute hack until AdoptOpenJdk provides Apple aarch64 builds
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java:138]: Which should be pretty soon from what I understand.

**Review by breskeby — APPROVED** [review]
nice, though I cannot test this locally

**Comment by droberts195** [comment]
> fails because it can't run the ML subprocess

elastic/ml-cpp#1651 contains the changes needed to get ML to compile on this platform.  The most difficult thing will be working out how to integrate this into CI and release manager builds, so it won't necessarily be easy to obtain artifacts that you don't build yourself in the near future.  But at least that PR proves there are no fundamental blockers to porting the ML code.

**Comment by pugnascotia** [comment]
Backported to `7.x` in fad1e7c02fd

**Pull request merged**